crates/orchestrator/src/workers/data_submission.rs

```diff
@@ -0,0 +1,55 @@
+use crate::config::config;
+use crate::jobs::create_job;
+use crate::jobs::types::JobType;
+use crate::workers::Worker;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use std::error::Error;
+
+pub struct DataSubmissionWorker;
+
+#[async_trait]
+impl Worker for DataSubmissionWorker {
+    // 0. All ids are assumed to be block numbers.
+    // 1. Fetch the latest completed Proving job.
+    // 2. Fetch the latest DA job creation.
+    // 3. Create jobs from after the lastest DA job already created till latest completed proving job.
+    async fn run_worker(&self) -> Result<(), Box<dyn Error>> {
+        let config = config().await;
+
+        // provides latest completed proof creation job id
+        let latest_proven_job_id = config
+            .database()
+            .get_last_successful_job_by_type(JobType::ProofCreation)
```

Comment:
We can delete this function `get_last_successful_job_by_type` from the mongo db functions. Instead we can use `get_last_successful_job_by_type` that is doing the same thing.